### PR TITLE
Inapp tracking calls 2

### PR DESCRIPTION
**Pull request by paulwsmith**
Adds manual tracking calls for custom in-app messages. These pass through to the respsective functionality in the native Android/iOS SDKs:
* `trackInAppMessageImpression(actionToken: string)`, 
* `trackInAppMessageClick(actionToken: string, parameters: Record<string, any> = {})` 
* `trackInAppMessageDismiss(actionToken: string)`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +38,29 @@ await pushNotificationReceived(snapyrData);
...
 await pushNotificationTapped(snapyrData);
+
+// --- In-App messaging section (optional) ---
+
+// listen to in-app messages from Snapyr 
+onSnapyrInAppMessage((inappMessage: SnapyrInAppMessage) => {
+    const actionToken = inappMessage.actionToken;
+    // example... stash this message in a useState so we can read its properties elsewhere
+    setCurrentInAppMessage(inappMessage);
+    if (inappMessage.actionType == "custom" && inappMessage.content.payloadType == "html") {
```
Comment (jonaldinger) [inline README.md:49]: At one point we had talked about the SDK automatically filtering out "non-custom" for the actions that get handed to the app. Did we decide against that? No biggie either way, just might be confusing to see an example here of filtering on custom if they will never see any other actionType.

File: README.md [hunk 2]
```diff
@@ -28,6 +38,29 @@ await pushNotificationReceived(snapyrData);
...
+    }
+});
+
+// track how users are interacting with your in-app message
+// ... after we've shown a message to the user...
+trackInAppMessageImpression(currentInAppMessage.actionToken);
+// ... after we've determined the user interacted with our message...
+trackInAppMessageClick(currentInAppMessage.actionToken, {exampleExtraProperty: "someId"});
+// ... if the user dismissed our message without interacting with it...
+trackInAppMessageDismiss(currentInAppMessage.actionToken);
```
Comment (jonaldinger) [inline README.md:61]: Does the app have to put in its own logic to only call this if no interactions happened prior? If so might be worth making that more explicit (either here or in the actual documentation).

**Review by jonaldinger — APPROVED** [review]
2 very minor comments on the readme, but LGTM!

File: README.md (hunk 1 above)
Comment (paulwsmith) [reply README.md:49]: That's right, good catch! Only custom actions trigger the callback. I'll update the example

File: README.md (hunk 2 above)
Comment (paulwsmith) [reply README.md:61]: Yes - the tracking calls (will) have more details in the documentation but I'll try to clarify the comment here.

File: src/index.tsx [hunk 3]
```diff
@@ -40,12 +40,26 @@ export type SnapyrConfigOptions = {
...
+  Overlay = 'overlay',
+}
+
+export enum SnapyrInAppPayloadType {
+  JSON = 'json',
+  HTML = 'html',
+}
+
+export declare type SnapyrInAppContent =
+  | { payloadType: SnapyrInAppPayloadType.JSON; payload: Record<string, any> }
```
Comment (jonaldinger) [inline src/index.tsx:54]: The leading pipe here is confusing to me. Is that a formatting thing or actual syntax?

**Review by jonaldinger — APPROVED** [review]
@paulwsmith already approved, but updates look good, thanks. One question just for my own curiosity.

File: src/index.tsx (hunk 3 above)
Comment (paulwsmith) [reply src/index.tsx:54]: Thanks @jonaldinger ! Just a formatting thing that was auto applied by the linter. This seems to be a fairly standard linting rule (we have it in the CP repo as well). I agree it's weird but I guess it's similar to including a trailing comma on the last line of a map / list - lets you reorder the lines later, if you want, without changing characters.

Comment (jonaldinger) [reply src/index.tsx:54]: Cool, TIL. Thanks!

**Pull request merged**